### PR TITLE
Add @grafana/ui CodeEditor behind a feature flag

### PR DESCRIPTION
**Pull request by andresmgot**
Starting point for #224 

This PR adds the new code editor behind a feature flag. Right now the editor has no additional functionality (apart from syntax highlighting) but those can be added incrementally without affecting possible releases.

I have also added some basic tests and mocks.

### CODE REVIEW

File: src/module.ts [hunk 1]
```diff
@@ -1,14 +1,12 @@
...
-import ConfigEditor from 'components/ConfigEditor';
 
-export const plugin = new DataSourcePlugin<
-  DataSourceApi<KustoQuery, AdxDataSourceOptions>,
-  KustoQuery,
-  AdxDataSourceOptions,
-  AdxDataSourceSecureOptions
->(AdxDataSource as any)
+export const plugin = new DataSourcePlugin<AdxDataSource, KustoQuery, AdxDataSourceOptions, AdxDataSourceSecureOptions>(
+  AdxDataSource as any
```
Comment (sunker) [inline src/module.ts:9]: Remove `as any`?

**Review by sunker — APPROVED** [review]
Nice!

File: src/module.ts (hunk 1 above)
Comment (sunker) [inline src/module.ts:9]: Remove `as any`?

**Review by sunker — APPROVED** [review]
Nice! Do we have an idea of what kind of issues we'll be running into when using the CodeEditor? Were these issues closed when we reverted back to the legacy editor a while back?

**Review by yaelleC — APPROVED** [review]
LGTM
That new editor would make my e2e tests a lot easier to set up 😉

File: src/module.ts (hunk 1 above)
Comment (andresmgot) [reply src/module.ts:9]: missed this. Thanks!

**Comment by andresmgot** [comment]
> Nice! Do we have an idea of what kind of issues we'll be running into when using the CodeEditor? Were these issues closed when we reverted back to the legacy editor a while back?

Yes, explained here: https://github.com/grafana/azure-data-explorer-datasource/issues/284. Once I investigate that more I will sanitize the feedback in bullet points in https://github.com/grafana/azure-data-explorer-datasource/issues/224

**Pull request merged**